Kumi.Game/Overlays/Control/ProgressNotification.cs

```diff
@@ -0,0 +1,164 @@
+﻿using Kumi.Game.Graphics;
+using osu.Framework.Allocation;
+using osu.Framework.Bindables;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Shapes;
+
+namespace Kumi.Game.Overlays.Control;
+
+public abstract partial class ProgressNotification : BasicNotification
+{
+    public BindableInt Current { get; set; } = new BindableInt();
+    public BindableInt Target { get; set; } = new BindableInt();
+    
+    /// <summary>
+    /// Whether the notification should close automatically when the progress is complete.
+    /// </summary>
+    public virtual bool AutoCloseUponCompletion { get; set; } = true;
+    
+    /// <summary>
+    /// Whether the notification can be canceled by the user.
+    /// </summary>
+    public virtual bool Cancellable { get; set; }
+    
+    /// <summary>
+    /// Whether the progress is complete.
+    /// </summary>
+    public bool IsFinished => Current.Value == Target.Value;
+
+    /// <summary>
+    /// An event that is fired when the progress is complete. Does not fire if the notification is canceled.
+    /// </summary>
+    public event Action? Finished;
+    
+    /// <summary>
+    /// An event that is fired when the notification is canceled.
+    /// </summary>
+    public event Action? Canceled;
+
+    public ProgressNotification(int target, int current = 0)
+    {
+        Current.Set(current);
+        Target.Set(target);
+        
+        _current = new BindableNumberWithCurrent<float>()
+        {
+            MinValue = 0,
+            MaxValue = target
+        };
+        
+        Current.BindValueChanged(_ => updateProgress(), true);
+        Target.BindValueChanged(_ =>
+        {
+            _current.MaxValue = Target.Value;
+            updateProgress();
+        }, true);
+    }
+    
+    public void Increment(int amount = 1) => Current.Set(Current.Value + amount);
+    public void Decrement(int amount = 1) => Current.Set(Current.Value - amount);
+    public void Set(int value) => Current.Set(value);
```

Comment:
Progress notifications shouldn't be handling where the current progress is. The consumer should be providing a bindable that would get bound in this class, and then changed accordingly outside of this progress notification.

Reply:
...I mean, it technically isn't tracking it's own progress, because the notification is completely useless without something that moves the bar or changes it's progress.

[The argument otherwise of not handling it fails, because osu!lazer accomplishes the same thing without requiring a bindable at all.](https://github.com/ppy/osu/blob/ee00f7ea1f6296dace5e7f15f6d9c8d95adaa23c/osu.Game/Database/RealmArchiveModelImporter.cs#L100-L132)

There's only a `Bindable` as a necessity solely because of how the current bar was designed around `SliderBar` as a convenience.

Reply:
We're not using a `SliderBar` anymore, however...? We may as well remove the bindable entirely then.